lib/src/install.rs

```diff
@@ -678,20 +678,27 @@ async fn initialize_ostree_root(
         ostree_ext::fsverity::ensure_verity(repo).await?;
     }
 
-    let stateroot_exists = rootfs_dir.try_exists(format!("ostree/deploy/{stateroot}"))?;
-    ensure!(
-        !stateroot_exists,
-        "Cannot redeploy over extant stateroot {stateroot}"
-    );
-    sysroot
-        .init_osname(stateroot, cancellable)
-        .context("initializing stateroot")?;
-
     let sysroot_dir = crate::utils::sysroot_dir(&sysroot)?;
 
     state.tempdir.create_dir("temp-run")?;
     let temp_run = state.tempdir.open_dir("temp-run")?;
 
+    {
+        let sysroot = SysrootLock::new_from_sysroot(&sysroot).await?;
+        let sysroot_storage = Storage::new(sysroot, &temp_run)?;
+        crate::deploy::cleanup(&sysroot_storage).await?;
+    }
+
+    if let Some(booted) = sysroot.booted_deployment() {
+        if stateroot == booted.stateroot() {
+            anyhow::bail!("Cannot redeploy over booted stateroot {stateroot}");
+        }
+    }
+
+    sysroot
+        .init_osname(stateroot, cancellable)
```

Comment:
Ah can you try just making this function call idempotent? Arguably actually it should be so in libostree... but anyways for now basically just:

```
let stateroot_path = format!("ostree/deploy/{stateroot}");
if !sysroot_dir.try_exists(stateroot_path)? {
  sysroot.init_osname(...)
}
``` 

or so? Then I don't think we need to do any extra cleanup in the `cleanup` function.

Reply:
nice, thanks! That's simpler